modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/source/KinesisEnv.scala

```diff
@@ -0,0 +1,151 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.postgres.source
+
+import java.util.{Base64, UUID}
+import java.net.InetAddress
+import java.nio.charset.StandardCharsets
+
+import cats.implicits._
+
+import cats.effect.{Blocker, ConcurrentEffect, ContextShift, Resource, Sync, Timer}
+
+import fs2.{Stream, Pipe}
+import fs2.aws.kinesis.{CommittableRecord, Kinesis}
+
+import software.amazon.awssdk.regions.Region
+import software.amazon.kinesis.common.ConfigsBuilder
+import software.amazon.kinesis.coordinator.Scheduler
+import software.amazon.kinesis.metrics.MetricsLevel
+import software.amazon.kinesis.processor.ShardRecordProcessorFactory
+import software.amazon.kinesis.retrieval.polling.PollingConfig
+import software.amazon.kinesis.retrieval.fanout.FanOutConfig
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient
+
+import com.snowplowanalytics.snowplow.badrows.{BadRow, Payload}
+import com.snowplowanalytics.snowplow.analytics.scalasdk.ParsingError.NotTSV
+
+import com.snowplowanalytics.snowplow.postgres.config.Cli
+import com.snowplowanalytics.snowplow.postgres.streaming.SinkPipe
+import com.snowplowanalytics.snowplow.postgres.streaming.data.BadData
+import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig.{Monitoring, Source, Purpose}
+
+object KinesisEnv {
+
+  def create[F[_]: ConcurrentEffect: ContextShift: Timer](blocker: Blocker, config: Source.Kinesis, metrics: Monitoring.Metrics, purpose: Purpose): Resource[F, Environment[F, CommittableRecord]] =
+    for {
+      kinesisClient <- makeKinesisClient[F](config.region)
+      dynamoClient <- makeDynamoDbClient[F](config.region)
+      cloudWatchClient <- makeCloudWatchClient[F](config.region)
+      kinesis = Kinesis.create(blocker, scheduler(kinesisClient, dynamoClient, cloudWatchClient, config, metrics, _))
+    } yield Environment(
+        getSource(kinesis),
+        getPayload(purpose),
+        checkpointer(kinesis, config.checkpointSettings),
+        SinkPipe.OrderedPipe.forTransactor[F]
+      )
+
+  private def getSource[F[_]](kinesis: Kinesis[F]): Stream[F, CommittableRecord] =
+    kinesis.readFromKinesisStream("THIS DOES NOTHING", "THIS DOES NOTHING")
```

Comment:
I wonder, where is `THIS DOES NOTHING` coming from? If it does nothing, why do we have it? maybe a small doc could help future maintainer

Reply:
These arguments are used to create `KinesisConsumerSettings` [in here](https://github.com/laserdisc-io/fs2-aws/blob/main/fs2-aws-kinesis/src/main/scala/fs2/aws/kinesis/Kinesis.scala#L75). Later on, `KinesisConsumerSettings` object created in here is used [in this function](https://github.com/laserdisc-io/fs2-aws/blob/main/fs2-aws-kinesis/src/main/scala/fs2/aws/kinesis/Kinesis.scala#L75) and only `bufferSize` field of given `KinesisConsumerSettings` object is used therefore given stream name and app name is not used in anywhere. I will write some small explanation as comment 👍